### PR TITLE
SNOW-1061848: Fix unit tests

### PR DESCRIPTION
**Pull request by sfc-gh-akowalczyk**
# Overview

SNOW-1061848

This PR fixes the unit tests for non-buffered scenario - some of them used to still validate the buffered scenario even if it was not expected.

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java [hunk 1]
```diff
@@ -252,7 +252,7 @@ public DirectTopicPartitionChannel(
             metricsJmxReporter,
             this.offsetPersistedInSnowflake,
             this.processedOffset,
-            new AtomicLong(-1));
+            new AtomicLong(0));
```
Comment (sfc-gh-akowalczyk) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java:255]: I need to change it back to conform with the expected behavior in tests.

File: src/test/java/com/snowflake/kafka/connector/internal/streaming/BufferedTopicPartitionChannelTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
```
Comment (sfc-gh-lshcharbaty) [inline src/test/java/com/snowflake/kafka/connector/internal/streaming/BufferedTopicPartitionChannelTest.java:33]: Let's use junit5 only in new classes

**Review by sfc-gh-lshcharbaty — APPROVED** [review]
lgtm

File: src/test/java/com/snowflake/kafka/connector/internal/streaming/BufferedTopicPartitionChannelTest.java (hunk 2 above)
Comment (sfc-gh-akowalczyk) [reply src/test/java/com/snowflake/kafka/connector/internal/streaming/BufferedTopicPartitionChannelTest.java:33]: That are the tests just moved from the previous file, I would prefer avoid refactoring setup to the latest version at this moment. Eventually, we will, hopefully, move all of them to JUnit 5.

**Pull request merged**